roles/tackle/defaults/main.yml

```diff
@@ -144,8 +144,14 @@ ui_container_limits_cpu: "500m"
 ui_container_limits_memory: "800Mi"
 ui_container_requests_cpu: "100m"
 ui_container_requests_memory: "350Mi"
+# Setting ingress defaults for 'nginx' ingress
+# To use 'alb' ingress need to set:
+#   ui_ingress_class_name: "alb"
+#   ui_ingress_path_type: "Prefix"
 ui_ingress_name: "{{ app_name }}"
+ui_ingress_class_name: "nginx"
 ui_ingress_proxy_body_size: "500m"
+ui_ingress_path_type: ImplementationSpecific
```

Comment:
You could do something like this so that the only value that has to be changed is the class name. If that doesn't cover some other unknown scenario it should remain possible to override it from the CR since extra vars have higher precedence than defaults here.
```
ui_ingress_path_type: "{{ 'Prefix' if ui_ingress_class_name == 'alb' else 'ImplementationSpecific' }}"
```

If Prefix/ImplementationSpecific combined with alb/nginx are the only two valid values it could even be written one way or the other (and expanded for other cases as well) with a conditional in the template, but I am fine with it either way.

From experiences with mig-controller, I don't think most people know they can look at defaults/main.yml when looking for options to change behavior, so I don't know if they'll see the comment to change both together (and we probably need to document that alb is an option somewhere (but I don't know if we have any actively updated documentation so that can be a separate PR as far as I am concerned).

Reply:
Open to doing this now, I imagine in future we will have other Ingresses that need to be supported but I don't know what they would be today.

I imagine at some point we may need to open up the knobs for these various tweaks or defer to the user to create their own Ingress.   I'm not sure how manageable it is for the Operator to be aware of a growing number of Ingress and the tweaks needed.

I could see going simple for now, just have 'nginx' and 'alb' and come back to this later when more are known.

Thoughts?

Reply:
Ya, I think if the number grows it get ugly doing it in defaults/main.yml, which is why I brought up the option of doing it in the template.

We should be able to do it in the template with conditionals and make it less ugly (and even make it possible to override in the event one of the configurations we don't know about is somehow possible just by changing this).

[Here](https://github.com/konveyor/tackle2-operator/pull/197/files#diff-bfde856d391d2cee80565372bec171ad04a4efa703c99c53ba92b8b790cfb30dR28)

you could do something like this, not define ui_ingress_path_type in defaults/main.yml then they can flip between alb and nginx with one var. If there are other cases covered by setting ui_ingress_path_type that can be done and/or we can add them to the conditional.

```
{% if ui_ingress_path_type is defined %}
            pathType: {{ ui_ingress_path_type }}
{% elif ui_ingress_class_name == 'alb' %}
            pathType: Prefix
{% else %}
            pathType: ImplementationSpecific
{% endif %}
```